x-pack/plugins/screenshotting/server/screenshots/observable.ts

```diff
@@ -173,15 +164,9 @@ export class ScreenshotObservableHandler {
     const waitTimeout = toNumber(this.config.capture.timeouts.waitForElements);
 
     return defer(() => getNumberOfItems(driver, this.eventLogger, waitTimeout, this.layout)).pipe(
-      mergeMap(async (itemsCount) => {
-        // set the viewport to the dimensions from the job, to allow elements to flow into the expected layout
-        const viewport = this.layout.getViewport(itemsCount) || getDefaultViewPort();
-
-        // Set the viewport allowing time for the browser to handle reflow and redraw
-        // before checking for readiness of visualizations.
-        await driver.setViewport(viewport, this.eventLogger.kbnLogger);
```

Comment:
The setViewport call has been moved to right before the screenshot capture, and it takes into account the dimensions of the element we're trying to capture.